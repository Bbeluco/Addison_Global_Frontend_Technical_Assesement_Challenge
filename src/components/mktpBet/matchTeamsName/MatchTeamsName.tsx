import React from "react";
import { StyleSheet, View, Text } from "react-native";

type MatchTeamsNameType = {
    eventName: string
}

function MatchTeamsName({ eventName }: MatchTeamsNameType): React.JSX.Element {
    return (
        <View style={styles.match}>
            <Text style={styles.matchInfo}>{eventName.split(' vs ')[0]}</Text>
            <Text style={styles.matchInfo}>VS</Text>
            <Text style={styles.matchInfo}>{eventName.split(' vs ')[1]}</Text>
        </View>
    )
}

const styles = StyleSheet.create({
    match: {
        borderWidth: 1,
        borderColor: 'black',
        flexDirection: 'row',
        justifyContent: 'space-between',
        padding: '10%'
    },
    matchInfo: {
        fontWeight: 'bold'
    }
})

export default MatchTeamsName