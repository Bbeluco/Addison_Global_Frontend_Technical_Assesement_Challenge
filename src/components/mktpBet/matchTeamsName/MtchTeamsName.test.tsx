import MatchTeamsName from "./MatchTeamsName";
import renderer from 'react-test-renderer';

describe('<MatchTeamsName />', () => {
    it('should render the component correctly', () => {
        const component = renderer.create(<MatchTeamsName eventName="Vasco vs 15 de Piracicaba"/>).toJSON();
        expect(component).toMatchSnapshot();
    })
})