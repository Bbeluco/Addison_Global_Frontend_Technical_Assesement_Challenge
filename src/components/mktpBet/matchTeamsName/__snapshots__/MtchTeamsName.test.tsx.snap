// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MatchTeamsName /> should render the component correctly 1`] = `
<View
  style={
    {
      "borderColor": "black",
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": "10%",
    }
  }
>
  <Text
    style={
      {
        "fontWeight": "bold",
      }
    }
  >
    Vasco
  </Text>
  <Text
    style={
      {
        "fontWeight": "bold",
      }
    }
  >
    VS
  </Text>
  <Text
    style={
      {
        "fontWeight": "bold",
      }
    }
  >
    15 de Piracicaba
  </Text>
</View>
`;
